hazelcast/src/main/java/com/hazelcast/instance/Node.java

```diff
@@ -252,6 +253,11 @@ private void initializeListeners(Config config) {
                 hazelcastInstance.lifecycleService.addLifecycleListener((LifecycleListener) listener);
                 known = true;
             }
+            if (listener instanceof ClientListener) {
+                String serviceName = ClientEngineImpl.SERVICE_NAME;
+                nodeEngine.getEventService().registerListener(serviceName, serviceName, listener);
```

Comment:
This one should be register local listener. 

A second thing, it would be better if we can reuse the following method here.
https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/client/impl/ClientServiceProxy.java#L48

But it does not seem straightforward to me because of circular dependency.